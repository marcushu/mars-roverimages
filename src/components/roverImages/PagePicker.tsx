import { FunctionComponent, useEffect, useState } from "react";
import styles from './pagePicker.module.css';

interface PagePickerProps {
  numOfPages: number
  currentPage: number
  setCurrentPage: React.Dispatch<React.SetStateAction<number>>
}

const PagePicker: FunctionComponent<PagePickerProps> = ({ numOfPages, currentPage, setCurrentPage }) => {
  const [pages, setpages] = useState<number[]>(); 

  useEffect(() => { 
    const numberArray = Array.from(Array(numOfPages + 1).keys()).splice(1);

    setpages(numberArray);
  }, [numOfPages]);

  return (
    <div className={styles.pagepicker} data-testid="pagepicker">
      {pages?.map(page => 
        <span 
          key={page}
          onClick={() => setCurrentPage(page)} 
          className={page === currentPage ? styles.selected : styles.pagepickernumber}>
          {page}
        </span>)}
    </div>
  );
}

export default PagePicker;