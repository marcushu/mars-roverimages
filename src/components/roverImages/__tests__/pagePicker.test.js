import { render, screen } from "@testing-library/react"
import PagePicker from "../PagePicker"

const mockSetPageFunc = jest.fn();

it('should display a number for each page', () => {
    render(<PagePicker currentPage={2} numOfPages={5} setCurrentPage={mockSetPageFunc} />);

    const view = screen.getByTestId('pagepicker')
    expect(view).toBeInTheDocument();
});

it('should call update function with the correct argument', () => {
    render(<PagePicker currentPage={2} numOfPages={5} setCurrentPage={mockSetPageFunc} />);

    const el = screen.getByText(/2/);
    
    el.click();
    expect(mockSetPageFunc.mock.calls[0][0]).toBe(2);
});

it('should match snapshot', () => {
    const { asFragment } = render(<PagePicker currentPage={2} numOfPages={5} setCurrentPage={mockSetPageFunc} />);

    expect(asFragment()).toMatchSnapshot();
});
