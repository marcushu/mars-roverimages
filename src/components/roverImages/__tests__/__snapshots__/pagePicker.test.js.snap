// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="pagepicker"
    data-testid="pagepicker"
  >
    <span
      class="pagepickernumber"
    >
      1
    </span>
    <span
      class="selected"
    >
      2
    </span>
    <span
      class="pagepickernumber"
    >
      3
    </span>
    <span
      class="pagepickernumber"
    >
      4
    </span>
    <span
      class="pagepickernumber"
    >
      5
    </span>
  </div>
</DocumentFragment>
`;
